lib/Lib.IntTypes.fst

```diff
@@ -60,6 +84,24 @@ let size_to_uint64 x = FStar.Int.Cast.uint32_to_uint64 x
 
 let byte_to_uint8 x = x
 
+let byte_to_int8 x = FStar.Int.Cast.uint8_to_int8 x
+
+let nat_to_int #t #l x =
+  match t with
+  | U1 -> u8 x
+  | U8 -> u8 x
+  | U16 -> u16 x
+  | U32 -> u32 x
+  | U64 -> u64 x
+  | U128 -> UInt128.uint_to_t x
+  | S8 -> i8 x
+  | S16 -> i16 x
+  | S32 -> i32 x
+  | S64 -> i64 x
+  | S128 -> Int128.int_to_t x
+  
+#reset-options "--z3rlimit 500"
```

Comment:
This is excessive. Note that this is running with a quite high `fuel` and `ifuel` and it will succeed with lower fuel and less resources.